src/main/java/com/singh/astha/notification/service/dtos/response/wrapper/ErrorResponse.java

```diff
@@ -2,28 +2,20 @@
 
 import com.fasterxml.jackson.annotation.JsonInclude;
 import com.singh.astha.notification.service.enums.ErrorCode;
-import lombok.*;
+import lombok.Builder;
 
 import java.io.Serializable;
 
-@Getter
-@Setter
 @Builder
-@AllArgsConstructor(access = AccessLevel.PROTECTED)
 @JsonInclude(JsonInclude.Include.NON_NULL)
-public class ErrorResponse implements Serializable {
-    private String errorCode;
+public record ErrorResponse(String errorCode, String message, String detail, String help) implements Serializable {
 
-    private String message;
-
-    private String detail;
-
-    private String help;
-
-    public static ErrorResponse from(ErrorCode errorCode) {
+    public static ErrorResponse from(ErrorCode errorCode, String detail) {
         return ErrorResponse.builder()
                 .errorCode(errorCode.getCode())
                 .message(errorCode.getMessage())
+                .detail(detail)
                 .build();
     }
+
```

Comment:
Adding detail here will require you pass null whenever it is not required.

Create overloads for this method.
(ErrorCode errorCode)
(ErrorCode errorCode, String detail)
(ErrorCode errorCode, String detail, String help)

Reply:
fixed